### ISSUE TITLE
[Feature Suggestion] Default effect times overhaul?

### ISSUE THREAD

**Issue by SeanMirrsen** [gta-chaos-mod/ChaosModV#593]
I've noticed that in the code, an effect is simply defined as either long, or short. This no doubt simplifies a great many things, especially in terms of settings, but as new effects are added, I think it's becoming clear that the two-step default gradation is no longer adequate. Some effects need to be shorter than the default 'short', some longer than the default 'long'. There's various ways to tackle this problem, however adding excessive complexity would also not be a welcome addition, I would expect. It's already possible to set custom timer lengths, however the default effect database (in EffectsInfo.h) don't seem to support that functionality.

Possible solutions:
- Expand the range of effect lengths, by one or two more tiers. Some effects need to be shorter than the default 'short', some longer than the default 'long'. Adding some more gradation won't make the settings menu much more complex, but it would still lack some flexibility as the new timer values would still be preset. The data structure would need to be changed to accept more than a 'true/false' for effect length.
- Change the data structure in EffectsInfo.h to be able to accept custom timer lengths, thus allowing effects to be of arbitrary preset length by default, should the effect require it. This would mirror the custom timer length value that can be set in the individual effect settings, but prevent those effects from being scaled with the short/long effect length settings, requiring the user to find the effects in the individual effects list to alter them.
- Add a new "timer ratio" variable to the data structure, allowing an effect to be set to a custom length proportional to either the short or the long timer length. This would complicate individual effect settings a bit more, as well as the data structure and many other things, but would allow both custom default effect lengths, and scaling those effect lengths using just the short/long effect length values in the settings.

Context for suggestion:
Some possible effects, like the "haywire teleport" I suggested, would be very disruptive - not to mention taxing on the game and the player - if they were to persist for the 'standard' short effect length. Some others, like activating some effect on every effect activation, don't fit with metaeffects but require exceedingly long effect length to trigger on several consecutive effects. I think this change would work to effectively tackle those problems, and allow more flexibility in the possible effects without them needing to be tweaked by the player.

**Comment by OHSHIET** [comment]
I agree. If you go through my suggestions here, some of them also have a problem with long/short effect durations or something like that. I suggested a metaeffect which would randomize the timer lengths #512, but this isn't what we actually want I guess
